src/FilledEntity.ts

```diff
@@ -38,11 +38,26 @@ export const getEntityDisplayValueMap = (filledEntityMap: FilledEntityMap): Map<
   return Object.keys(filledEntityMap.map).reduce((m, key) => {
     const entityDisplayValue = filledEntityMap.ValueAsString(key)
 
-    // TODO: Required check because poor API from filledEntityMap which can return null
     if (entityDisplayValue) {
       m.set(key, entityDisplayValue)
     }
 
+    // Entity is missing. Provide readable error message
+    else {
+      // Default to showing key
+      let missingValue = key
+
+      // If key is an entityId, try to get value from name key
+      if (filledEntityMap.map[key].entityId ===  key) {
+        for (let k of Object.keys(filledEntityMap.map)) {
```

Comment:
Looks like a `find` operation

Reply:
find operates on arrays.  This is operating on a key / values.   could transform it to an array first, but this seems as good

Reply:
You did transform it to an array with `Object.keys`  that's the point of the method.